CI: Conformance Runtime (ci-runtime): Test_NeighborAddDel: Received unexpected error: did not receive expected peering state ["ESTABLISHED"], context deadline exceeded

## CI failure

```
--- FAIL: Test_NeighborAddDel (60.06s)
    --- FAIL: Test_NeighborAddDel/add_two_neighbors (59.97s)
        neighbor_test.go:153: 
            	Error Trace:	/host/pkg/bgpv1/test/neighbor_test.go:153
            	Error:      	Received unexpected error:
            	            	did not receive expected peering state ["ESTABLISHED"], context deadline exceeded
            	Test:       	Test_NeighborAddDel/add_two_neighbors
            	Messages:   	add two neighbors
    --- FAIL: Test_NeighborAddDel/update_both_neighbors (0.00s)
        neighbor_test.go:153: 
            	Error Trace:	/host/pkg/bgpv1/test/neighbor_test.go:153
            	Error:      	Received unexpected error:
            	            	did not receive expected peering state ["ESTABLISHED"], context deadline exceeded
            	Test:       	Test_NeighborAddDel/update_both_neighbors
            	Messages:   	update both neighbors
    --- FAIL: Test_NeighborAddDel/delete_both_neighbors (0.00s)
        neighbor_test.go:153: 
            	Error Trace:	/host/pkg/bgpv1/test/neighbor_test.go:153
            	Error:      	Received unexpected error:
            	            	did not receive expected peering state ["IDLE" "ACTIVE"], context deadline exceeded
            	Test:       	Test_NeighborAddDel/delete_both_neighbors
            	Messages:   	delete both neighbors
```

Link: https://github.com/cilium/cilium/actions/runs/8183062795/job/22375383468
Sysdump: No sysdump produced for unit tests

We had a similar issue before https://github.com/cilium/cilium/issues/25637 and https://github.com/cilium/cilium/pull/25641 fixed that, but this is failing even with the retry, so another issue.

--- a/pkg/bgpv1/test/objects.go
+++ b/pkg/bgpv1/test/objects.go
@@ -14,6 +14,7 @@ import (
 
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/utils/ptr"
 )
 
 var (
@@ -44,6 +45,17 @@ func newPolicyObj(conf policyConfig) v2alpha1.CiliumBGPPeeringPolicy {
 	}
 
 	if conf.virtualRouters != nil {
+		// Override ConnectRetryTimeSeconds to 1 for all neighbors
+		// unless it is specified explicitly. Otherwise, the default
+		// value of 120 seconds would timeout the tests when the
+		// initial connect fails.
+		for i, vrouter := range conf.virtualRouters {
+			for j, neigh := range vrouter.Neighbors {
+				if neigh.ConnectRetryTimeSeconds == nil {
+					conf.virtualRouters[i].Neighbors[j].ConnectRetryTimeSeconds = ptr.To[int32](1)
+				}
+			}
+		}
 		policyObj.Spec.VirtualRouters = conf.virtualRouters
 	}
 

